solver/src/problem/fitness.rs

```diff
@@ -232,13 +225,13 @@ impl JsspFitness {
                 // space to store only the direct predecessor (list of direct predecessors?).
                 if op.id != indv.operations.len() - 1 {
                     if let Some(direct_pred_id) = op.preds.last() {
-                        if finish_times[*direct_pred_id] as f64 > time as f64 + delay {
+                        if indv.operations[*direct_pred_id].finish_time.unwrap_or(usize::MAX) as f64 > time as f64 + delay {
```

Comment:
Previously when there was no value in `finish_times` array we would get `usize::MAX` by default. Consider keeping this behaviour or make educated decision about change.